joern-cli/frontends/javasrc2cpg/src/main/scala/io/joern/javasrc2cpg/astcreation/expressions/AstForNameExpressionsCreator.scala

```diff
@@ -65,28 +68,45 @@ trait AstForNameExpressionsCreator { this: AstCreator =>
     }
   }
 
+  def createImplicitBaseFieldAccess(
+    isStatic: Boolean,
+    baseTypeDecl: TypeDeclScope,
+    node: Node,
+    fieldName: String,
+    fieldTypeFullName: String
+  ): Ast = {
+    val base =
+      if (isStatic) {
+        NewTypeRef()
+          .code(baseTypeDecl.typeDecl.name)
+          .typeFullName(baseTypeDecl.typeDecl.fullName)
+          .lineNumber(line(node))
+          .columnNumber(column(node))
+      } else {
+        newIdentifierNode(NameConstants.This, baseTypeDecl.typeDecl.fullName)
+      }
+    createFieldAccessAst(
+      Ast(base),
+      s"${base.code}.${fieldName}",
+      line(node),
+      column(node),
+      fieldName,
+      fieldTypeFullName,
+      line(node),
+      column(node)
+    )
+  }
+
   private def astForStaticImportOrUnknown(nameExpr: NameExpr, name: String, typeFullName: Option[String]): Ast = {
     tryWithSafeStackOverflow(nameExpr.resolve()) match {
       case Success(value) if value.isField =>
-        val identifierName = if (value.asField.isStatic) {
-          // TODO: v is wrong. Statically imported expressions can also be represented by just the name.
-          // A static field represented by a NameExpr must belong to the class in which it's used. Static fields
-          // from other classes are represented by a FieldAccessExpr instead.
```

Comment:
This was very confusingly written. It's basically 2 comments in one. The original comment was
>           // A static field represented by a NameExpr must belong to the class in which it's used. Static fields
>           // from other classes are represented by a FieldAccessExpr instead.

which is wrong. Instead of removing this comment and replacing it with a todo, I decided (for some reason which I no longer remember or understand) to add the TODO

>  // TODO: v is wrong. Statically imported expressions can also be represented by just the name.

The idea I was trying to communicate with the TODO still applies, I think. My original assumption was that a name expression without  an explicit scope must refer to a field declared in the same class as the name expression, but that's not true because of static imports, as in the example below. We would actually need to check whether the name has been imported statically. I remember running into some issue when looking at this before, but not what that issue was. 

```
import static foo.Foo.FOO;

public class Bar {
  public static void main(String[] args) {
    System.out.println(FOO);
  }
}
```

I don't think fixing that should be a part of this PR, but keeping a TODO would be good, I think.